remotespark/livyclientlib/sparkcontroller.py

```diff
@@ -36,15 +38,15 @@ def run_cell(self, client_name, context, cell):
 
         # Execute in context
         if context == Constants.context_name_sql:
-            res = client_to_use.execute_sql(cell)
+            result = client_to_use.execute_sql(cell)
         elif context == Constants.context_name_hive:
-            res = client_to_use.execute_hive(cell)
+            result = client_to_use.execute_hive(cell)
         elif context == Constants.context_name_spark:
-            res = client_to_use.execute(cell)
+            result = client_to_use.execute(cell)
         else:
             raise ValueError("Context '{}' specified is not known.")
 
-        return res
+        return result.render(get_ipython())
```

Comment:
Do unit tests start throwing when multiple unit tests do a get_ipython call?
If so, you should wrap the call to get_ipython() in its own method so that it's mockable.